### PR TITLE
React & Redux  group project - Space Travelers' Hub 

### PR DESCRIPTION
**Pull request by Fadahunsiseyi**
### In this PR, We implemented  the following elements and steps:

  > - The Rockets section displays a list of all available Space rockets. Users can book each rocket by clicking the 
      reservation button or canceling the previously made booking.
  > - The Missions section displays a list of current missions along with their brief description and participation status.
  > - The Profile section displays all reserved rockets and space missions.
  > - Add reserved dragon list to my profile
  > - Add reservation to the dragon list
  > - Add styles to the dragon page
  > - Use React documentation.
  > - Use React components.
  > - Use React props.
  > - Use React Router.
  > - Connect React and Redux.
  > - Handle events in a React app.
  > - Write unit tests with React Testing Library.
  > - Use styles in a React app.
  > - Use React life cycle methods.
  > - Apply React best practices and language style guides in code.
  > - Use store, actions, and reducers in React.
  > - Perform a code review for a team member.
  > - Fix linters error

### CODE REVIEW

File: src/components/Mission.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    join,
+  } = props;
+  const dispatch = useDispatch();
+  const handleClick = () => {
+    if (join) {
+      dispatch(leaveMission(id));
+    } else {
+      dispatch(joinMission(id));
+    }
+  };
```
Comment (vechicin) [inline src/components/Mission.js:20]: Actions should only be dispatched once and they are being dispatched twice, please fix this.

Screen Shot 2022-08-26 at 8 17 41

Screen Shot 2022-08-26 at 8 18 10

File: src/components/Rocket.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+const Rock = (props) => {
+  const rocket = props;
+  const dispatch = useDispatch();
+  const { reserve } = rocket.rock;
+  const handleReserve = (e) => {
+    e.target.classList.toggle('reservedButton');
+    const toggle = !reserve;
+    const setReserve = (() => ({ ...rocket.rock, reserve: toggle }));
+    dispatch(createReserve(setReserve()));
+  };
```
Comment (vechicin) [inline src/components/Rocket.js:13]: Actions should only be dispatched once as mentioned above, please fix this.

Screen Shot 2022-08-26 at 8 18 59

Screen Shot 2022-08-26 at 8 19 15

File: src/components/Mission.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
```
Comment (vechicin) [inline src/components/Mission.js:3]: According to the project requirements, you should use `useSelector()` to get a slice of the `store` in the component. Please fix this.

Screen Shot 2022-08-26 at 8 21 18

File: src/components/Rocket.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { useDispatch } from 'react-redux';
```
Comment (vechicin) [inline src/components/Rocket.js:1]: According to the project requirements, you should use `useSelector()` to get a slice of the `store` in the component. Please fix this.

Screen Shot 2022-08-26 at 8 21 18

**Review by vechicin — CHANGES_REQUESTED** [review]
# Hi @Fadahunsiseyi, @ademibowale and @omarbabou ✋, 

Good job so far! 🔥 
There are some issues that you still need to work on to go to the next project but you are almost there! 🚨 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights ⚡ 

- [x] Good job implementing most of the required features for this project!

## Required Changes ♻️

- [ ] Since this is a group comprised of 3 people, you need to implement all of the required features for the `Dragons` page as well. Please do so.
- [ ] **NOTE TO REVIEWERS**. Since the students didn't implement features related to the `Dragons` page, feel free to review any change they introduce on that page and the updates necessary for the `My Profile` page as if it was the first review.

_Check the comments under the review._

## Optional suggestions 📂 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done working on the requested changes from the previous reviewer 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**